food-truck-frontend/src/pages/interactive-map.tsx

```diff
@@ -0,0 +1,25 @@
+import React, {useEffect, useState} from 'react'
+import {DEFAULT_ERR_RESP} from '../api/DefaultResponses';
+import {getNearbyTruckLocations} from '../api/TruckApi';
+import TruckLocationMapComponent from "../components/map/truck_location_map/TruckLocationMapComponent"
+import {RouteLocation} from "../domain/RouteLocation";
+
+function InteractiveMapPage() {
+  let [locations, setLocations] = useState<RouteLocation[]>([]);
+  let [ready, setReady] = useState<boolean>(false);
+  useEffect(() => {
+    getNearbyTruckLocations(DEFAULT_ERR_RESP)
+      .then(locs => {
+        if (locs) {
+          setLocations(locs);
+          setReady(true);
+        }
+      });
+    });
+    return(
+        ready && <TruckLocationMapComponent locations={locations} />
```

Comment:
I'd like to ask we switch this to `ready? <map> : "Please wait"`. I can see how the operator precedence works after double-taking, but it's not as clear as the ternary operator.